### DISCUSSION TITLE
error：Terminated workflow Why got：nDCHistoryReplicator unable to create workflow when applyStartEvents

### DISCUSSION THREAD

**Discussion by Johnite** [uber/cadence#4372] [Q&A]
我有个workflow，本来已经terminate了，但是在后台cadence-service的日志中仍然可以看到很多的error：nDCHistoryReplicator unable to create workflow when applyStartEvents。
如图：

但是我后台有很多的这种错误：
{"level":"error","ts":"2021-08-16T15:23:21.455+0800","msg":"nDCHistoryReplicator unable to create workflow when applyStartEvents","service":"cadence-history","shard-id":1590,"address":"IP1:7934","component":"history-replicator","wf-id":"serial_raw_feature_sql_219","wf-run-id":"4f07f4dd-d92c-43c7-b2f1-cf19ce9e49f5","xdc-source-cluster":"active","xdc-incoming-version":1,"wf-first-event-id":1,"wf-next-event-id":2,"error":"Workflow execution already running. WorkflowId: serial_raw_feature_sql_219","logging-call-at":"history_replicator.go:333","stacktrace":"github.com/uber/cadence/common/log/loggerimpl.(*loggerImpl).Error\n\t/Users/didi/go/src/go.uber.org/cadence-0.20.0/common/log/loggerimpl/logger.go:131\ngithub.com/uber/cadence/service/history/ndc.(*historyReplicatorImpl).applyStartEvents\n\t/Users/didi/go/src/go.uber.org/cadence-0.20.0/service/history/ndc/history_replicator.go:333\ngithub.com/uber/cadence/service/history/ndc.(*historyReplicatorImpl).applyEvents\n\t/Users/didi/go/src/go.uber.org/cadence-0.20.0/service/history/ndc/history_replicator.go:229\ngithub.com/uber/cadence/service/history/ndc.(*historyReplicatorImpl).ApplyEvents\n\t/Users/didi/go/src/go.uber.org/cadence-0.20.0/service/history/ndc/history_replicator.go:200\ngithub.com/uber/cadence/service/history.(*historyEngineImpl).ReplicateEventsV2\n\t/Users/didi/go/src/go.uber.org/cadence-0.20.0/service/history/historyEngine.go:2427\ngithub.com/uber/cadence/service/history/replication.(*taskExecutorImpl).handleHistoryReplicationTaskV2\n\t/Users/didi/go/src/go.uber.org/cadence-0.20.0/service/history/replication/task_executor.go:206\ngithub.com/uber/cadence/service/history/replication.(*taskExecutorImpl).execute\n\t/Users/didi/go/src/go.uber.org/cadence-0.20.0/service/history/replication/task_executor.go:92\ngithub.com/uber/cadence/service/history/replication.(*taskProcessorImpl).processTaskOnce\n\t/Users/didi/go/src/go.uber.org/cadence-0.20.0/service/history/replication/task_processor.go:426\ngithub.com/uber/cadence/service/history/replication.(*taskProcessorImpl).processSingleTask.func1.1\n\t/Users/didi/go/src/go.uber.org/cadence-0.20.0/service/history/replication/task_processor.go:381\ngithub.com/uber/cadence/common/backoff.Retry\n\t/Users/didi/go/src/go.uber.org/cadence-0.20.0/common/backoff/retry.go:105\ngithub.com/uber/cadence/service/history/replication.(*taskProcessorImpl).processSingleTask.func1\n\t/Users/didi/go/src/go.uber.org/cadence-0.20.0/service/history/replication/task_processor.go:373\ngithub.com/uber/cadence/common/backoff.Retry\n\t/Users/didi/go/src/go.uber.org/cadence-0.20.0/common/backoff/retry.go:105\ngithub.com/uber/cadence/service/history/replication.(*taskProcessorImpl).processSingleTask\n\t/Users/didi/go/src/go.uber.org/cadence-0.20.0/service/history/replication/task_processor.go:389\ngithub.com/uber/cadence/service/history/replication.(*taskProcessorImpl).processResponse\n\t/Users/didi/go/src/go.uber.org/cadence-0.20.0/service/history/replication/task_processor.go:300\ngithub.com/uber/cadence/service/history/replication.(*taskProcessorImpl).processorLoop\n\t/Users/didi/go/src/go.uber.org/cadence-0.20.0/service/history/replication/task_processor.go:207"}

为什么呢？哪位大神给指点一下？

**Comment by longquanzheng** [comment]
This is should be caused by some retry in replication. It should be safe to ignore unless you see some wrong states in replicated workflow
